cpp/src/stream_compaction/apply_boolean_mask.cu

```diff
@@ -67,20 +71,38 @@ std::unique_ptr<table> apply_boolean_mask(table_view const& input,
                                           column_view const& boolean_mask,
                                           rmm::cuda_stream_view stream,
                                           rmm::device_async_resource_ref mr)
+{
+  return apply_boolean_mask(input, boolean_mask, 0, input.num_rows(), stream, mr);
```

Comment:
Suggestion needed: Kept this detail API as it's being used directly at many places within libcudf detail namespace. If this seems like an anti-pattern, I have a local commit which removes this and replaces its use with the public version that I can push to this PR?